### PR TITLE
Enhancement to add support for classpath sources formatter configurations

### PR DESCRIPTION
**Pull request by matthewwerny**
Summary of changes:

- Added support project to contain code useful across project artifacts
- Added Resource class to provide abstraction for files obtained from
  the classpath or from the filesystem
- Corrected reference to missing html-core artifact that was causing the
  build to fail

### CODE REVIEW

File: support/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	<parent>
+		<groupId>com.marvinformatics.formatter</groupId>
+		<artifactId>parent</artifactId>
+		<version>1.5.0-SNAPSHOT</version>
+	</parent>
+
+	<artifactId>support</artifactId>
+	<packaging>jar</packaging>
+
+	<developers>
```
Comment (velo) [inline support/pom.xml]: you can remove <developers> from this pom, keep on parent only

File: support/pom.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	<modelVersion>4.0.0</modelVersion>
+
+	<parent>
+		<groupId>com.marvinformatics.formatter</groupId>
+		<artifactId>parent</artifactId>
+		<version>1.5.0-SNAPSHOT</version>
+	</parent>
+
+	<artifactId>support</artifactId>
+	<packaging>jar</packaging>
```
Comment (velo) [inline support/pom.xml]: not necessary, feel free to remove it

**Comment by velo** [comment]
My only concern is if the eclipse plugin still working, I will check that

**Comment by matthewwerny** [comment]
It should still work, but not support the enhanced syntax.  I'll follow up with an appropriate PR to account for that.

**Pull request merged**